tests/test_plot.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+import dgpost.utils
+import numpy as np
+import pandas as pd
+import pytest
+
+from tests.utils import datadir
+
+
+@pytest.mark.parametrize(
+    "input",
+    [
+        {
+            "style": "style",
+            "axes": [
+                {
+                    "lines": [
+                        {
+                            "y": "sin1",
+                            "color": "r",
+                        },
+                        {
+                            "y": "sin2",
+                            "color": "b",
+                        },
+                    ],
+                    "xlim": (0, 7),
+                },
+            ],
+            "show": True,
+            "save": "",
+        },
+    ],
+)
+def test_plot(input, datadir):
```

Comment:
Added a new more complex example.
I also now check the plots against a reference image, like you suggested.

Reply:
It should be already possible by adding the keyword `gridspec_kw={'width_ratios': [x,y], 'height_ratios': [a,b]}`.
I could make it more accessible, if you want.

Reply:
If you could add a test case that does it, that would be great.

Reply:
Since I added `gridspec`, you can define the `grid` on the figure layer with `nrows` and `ncols` and then select the height/width on the axes layer by the keywords `rows, cols` with a `tuple`.